diff --git a/bootable/recovery/mmcutils/Android.mk b/bootable/recovery/mmcutils/Android.mk
index f1fe294..bb78a3e 100644
--- a/bootable/recovery/mmcutils/Android.mk
+++ b/bootable/recovery/mmcutils/Android.mk
@@ -7,6 +7,10 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := \
 	mmcutils.c
 
+ifneq ($(TARGET_RECOVERY_FORCE_BACKUP_SIZE),)
+LOCAL_CFLAGS += -DIMAGESIZE=$(TARGET_RECOVERY_FORCE_BACKUP_SIZE)
+endif
+
 LOCAL_MODULE := libmmcutils
 LOCAL_MODULE_TAGS := eng
 
diff --git a/bootable/recovery/mmcutils/mmcutils.c b/bootable/recovery/mmcutils/mmcutils.c
index f9c7b70..430a306 100644
--- a/bootable/recovery/mmcutils/mmcutils.c
+++ b/bootable/recovery/mmcutils/mmcutils.c
@@ -489,8 +489,14 @@ mmc_raw_dump_internal (const char* in_file, const char *out_file) {
     if (out == NULL)
         goto ERROR2;
 
+#ifndef IMAGESIZE
     fseek(in, 0L, SEEK_END);
     sz = ftell(in);
+#else //for some reason nandroid backup couldn't obtain size of MTK emmc partitions(boot/recovery), 
+      //force pushing its whole size so that backup will go on.
+    sz = IMAGESIZE;
+#endif
+
     fseek(in, 0L, SEEK_SET);
 
     if (sz % 512)
